aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncClient.java

```diff
@@ -143,16 +144,36 @@ public static AppSyncClient via(@NonNull GraphQLBehavior api) {
         return new NoOpCancelable();
     }
 
+    /**
+     * Uses Amplify API to make a mutation which will only apply if the version sent matches the server version.
+     *
+     * @param model      An instance of the Model with the values to mutate
+     * @param version    The version of the model we have
+     * @param onResponse Invoked when response data is available.
+     * @param onFailure  Invoked on failure to obtain response data
+     * @return A {@link Cancelable} to provide a means to cancel the asynchronous operation
+     */
+    @NonNull
+    @Override
+    public <T extends Model> Cancelable update(
+            @NonNull T model,
+            @NonNull Integer version,
+            @NonNull Consumer<GraphQLResponse<ModelWithMetadata<T>>> onResponse,
+            @NonNull Consumer<DataStoreException> onFailure) {
+        return update(model, version, null, onResponse, onFailure);
+    }
+
     @SuppressWarnings("unchecked") // (Class<T>)
     @NonNull
     @Override
     public <T extends Model> Cancelable update(
             @NonNull T model,
             @NonNull Integer version,
+            @Nullable QueryPredicate predicate,
             @NonNull Consumer<GraphQLResponse<ModelWithMetadata<T>>> onResponse,
             @NonNull Consumer<DataStoreException> onFailure) {
         try {
-            final String doc = AppSyncRequestFactory.buildUpdateDoc(model.getClass());
+            final String doc = AppSyncRequestFactory.buildUpdateDoc(model.getClass(), predicate != null);
```

Comment:
it feels like `buildUpdateDoc` should take the `predicate` and not a `boolean` indicating there's a predicate. 

I guess ideally `buildUpdateDocument` should return a `GraphQLDocument` or something like that, instead a `String`, the it would build the `buildDocument` would be a self-contained document factory.

I understand this can be a big undertaking for the moment, so feel free to take this feedback for a future improvement (maybe create a ticket?)

Reply:
^ I agree with this.

Reply:
My original intent was to do just that. However, as I started going through the interactions with the API category, it looked like it was going to require a larger refactor. I'm going to create an issue so we can track this though.